File: commands/delete.js
import { green } from 'https://deno.land/std@0.83.0/fmt/colors.ts'

export const deleteLocalTask = (taskName, json, HOME) => {
    for (let i = 0; i < json.tasks.length; i++) {
        const task = json.tasks[i];

        if (task.name === taskName) {
            json.tasks.splice(i, 1)
        }
    }

    const data = JSON.stringify(json)

    // WRITE
    Deno.writeTextFileSync(`${HOME}/tascli.json`, data)
}

File: lib/commands/delete.dart
import 'dart:io';
import 'package:tascli/helpers/error.dart';
import 'package:tascli/helpers/json.dart';

void deleteLocalTask(Map<String, dynamic> json, String taskName) async {
  try {
    for (int i = 0; i < json['tasks'].length; i++) {
      if (json['tasks'][i]['name'] == taskName) {
        json['tasks'].removeAt(i);
        await writeJson(json);
        exit(200);
      }
    }
    taskNotFound();
  } catch (RangeError) {
    taskNotFound();
  }
}

